File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        //sliding window
        //tc - O(n)
        int res = 0;
        int l = 0;
        int r = 0;
        int z = 0;
        while(r<nums.length){
            if(nums[r]==0){
                z+=1;
            }
            if(z>k){
                if(nums[l]==0){
                    z-=1;
                }
                l+=1;
            }
            if(z<=k){
                int temp = r-l+1;
                res = Math.max(res,temp);
            }
            r+=1;
        }
        return res;
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        #sliding window
        res = 0
        l = 0
        r = 0
        z = 0
        while r<len(nums):
            if nums[r]==0:
                z +=1
            while z>k:
                if nums[l]==0:
                    z-=1
                l+=1
            temp = r-l+1
            res = max(res,temp)
            r+=1
        return res
        